storage/postgres/postgres.go

```diff
@@ -208,40 +81,40 @@ func (p *Postgres) ReadPage(ctx context.Context, store string, tupleKey *openfga
 	}
 	defer iter.Stop()
 
-	return iter.toArray(ctx, opts)
+	return iter.ToArray(ctx, opts)
 }
 
-func (p *Postgres) read(ctx context.Context, store string, tupleKey *openfgapb.TupleKey, opts storage.PaginationOptions) (*tupleIterator, error) {
-	ctx, span := p.tracer.Start(ctx, "postgres.read")
+func (p *Postgres) read(ctx context.Context, store string, tupleKey *openfgapb.TupleKey, opts storage.PaginationOptions) (*storage.SQLTupleIterator, error) {
```

Comment:
Can we keep this more generic and just return `storage.TupleIterator`?

Reply:
It doesn't seem so. `ReadPage` uses `.toArray` which is not defined on `storage.TupleIterator`.

Reply:
Seems like a good candidate to add to `storage.TupleIterator` or as a helper method `ToArray(iter storage.TupleIterator)`. Introducing another structure just to attach the `.toArray` method to is a bit overkill.